src/hotspot/share/opto/macro.cpp

```diff
@@ -1604,7 +1603,10 @@ void PhaseMacroExpand::expand_initialize_membar(AllocateNode* alloc, InitializeN
       // No InitializeNode or no stores captured by zeroing
       // elimination. Simply add the MemBarStoreStore after object
       // initialization.
-      MemBarNode* mb = MemBarNode::make(C, Op_MemBarStoreStore, Compile::AliasIdxBot);
+      // See comment below in this if's other branch that explains why a raw memory MemBar is good enough. If init is
+      // null, this allocation does have an InitializeNode but this logic can't locate it (see comment in
+      // PhaseMacroExpand::initialize_object()).
+      MemBarNode* mb = MemBarNode::make(C, Op_MemBarStoreStore, Compile::AliasIdxRaw);
```

Comment:
Would it not be nicer to have the explanation here than below, and refer from below to here? Would help the reading flow ;)